libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -234,12 +234,82 @@ void IRGeneratorForStatements::endVisit(Return const& _return)
 
 void IRGeneratorForStatements::endVisit(UnaryOperation const& _unaryOperation)
 {
-	if (type(_unaryOperation).category() == Type::Category::RationalNumber)
+	auto const& opType = type(_unaryOperation);
```

Comment:
```suggestion
	Type const& opType = type(_unaryOperation);
```

Reply:
Also this is the result type, which is a big difference for the `delete` operator - but not sure if that is important for this PR.